lvgl: Add config to disable monochrome pixel software inversion

vulporuza opened the pull request:
In some cases, pixel inversion is managed by the display hardware (e.g. SSD1306) and software inversion is not required (as it disables the hardware inversion). This PR adds a configuration to disable this inversion.

vulporuza committed:
shields: ssd1306: Disable LVGL software screen inversion

On this screen the inversion is managed by hardware.

Signed-off-by: Robin-Charles Guihéneuf <robin-charles@hotmail.fr>

uLipe commented:
The code looks good, please just check the failing tests.

pdgendt commented:
> The code looks good, please just check the failing tests.

Not sure if the CI failure is related? 🤔 Restarted the job to make sure.

kartben reviewed (dismissed):
Please mention the new Kconfig option in the release notes

kartben commented on modules/lvgl/Kconfig:
```diff
@@ -94,6 +94,10 @@ config LV_COLOR_16_SWAP
 	bool "Swap the 2 bytes of RGB565 color."
 	depends on LV_COLOR_DEPTH_16
 
+config LV_Z_COLOR_MONO_HW_INVERSION
+	bool "Disable software pixel inversion."
```
sorry if i'm missing something but isn't this Kconfig poorly named? Should one select it or *not* select it to *disable* the SW inversion? Looks like there is too much double negation happening? Putting as a blocking comment which I will be happy to remove if it is indeed just me who's missing something obvious.

As a note, the "In some cases, pixel inversion is managed by the display hardware (e.g. SSD1306) and software inversion is not required (as it disables the hardware inversion)" paragraph from your PR would be a really good `help` text for this Kconfig :)

vulporuza replied:
The Kconfig indicates that the hardware (the screen itself) implements the pixel inversion. Therefore no software inversion is required. Accordingly, the corresponding code is disabled.

This makes sense to me, but I can change it, what do you suggest?

From my research, this type of screen (supporting hw inversion) is not the common case, so the negative approach is to add the Kconfig to all but a few boards.

vulporuza replied:
@kartben The config text could be "Do not use software pixel inversion", does it make more sense to you ?

kartben replied:
I'm sorry, I completely missed your replies.
More than the help text, my concern is that the name of the symbol doesn't actually convey it's meant to be a "disable" flag and is not self explanatory. Right now you actually end up with code that looks odd and even looks like a bug at first sight IMO: `if(CONFIG_FOO_INVERSION) don't invert!! else invert"

So basically what I am suggesting is making this `LV_Z_COLOR_MONO_NO_HW_INVERSION` to align with what seems to be a [common naming scheme](https://docs.zephyrproject.org/latest/kconfig.html#!_no_) for "disable when Y" options.

Does that make sense?

kartben replied:
Ok, sorry that I'm slow. I get it now. 😶
So I think what's confusing is that the description talks about "disabling" something while the name of the option itself is referring to a hardware feature being present or not. I'll dismiss my review because I don't want to block this longer than I already have and don't have brilliant suggestions to help it make less confusing, and it might very well be just me anyway.

Maybe "Hardware pixel inversion (disables software inversion)?

vulporuza committed:
lvgl: Add config to disable monochrome pixel software inversion

In some cases, pixel inversion is managed by the display hardware
and software inversion is not required.

Signed-off-by: Robin-Charles Guihéneuf <robin-charles@hotmail.fr>
